### ISSUE TITLE
Puppet is not supported on Ruby 2.2

### ISSUE THREAD

**Issue by BanzaiMan** [travis-ci/dpl#206]
Puppet does not currently run on Ruby 2.2 because `Syck` is no longer available. https://travis-ci.org/travis-ci/dpl/jobs/43838016#L226

```
/home/travis/build/travis-ci/dpl/vendor/bundle/ruby/2.2.0/gems/puppet-3.7.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml/syck_node_monkeypatch.rb:42:in `<top (required)>': uninitialized constant Syck (NameError)
```

**Comment by anatol** [comment]
Ruby 2.2 is released as stable and now it is recommended version. Do you have plans to fix puppet?

**Comment by kylog** [comment]
Puppet 4.0 is tested against ruby 2.2 and should be released in the next few weeks.

**Comment by jdoshi1** [comment]
Is this issue resolved? I'm still not able to run Librarian-puppet. The following is error stack trace:

$ librarian-puppet install --no-use-v1-api
Replacing Puppet Forge API URL to use v3 https://forgeapi.puppetlabs.com. You should update your Puppetfile
Unable to load puppet. Please install it using native packages for your platform (eg .deb, .rpm, .dmg, etc).
puppet --version returned pid 37144 exit 1
/usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml/syck_node_monkeypatch.rb:42:in `<top (required)>': uninitialized constant Syck (NameError)
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:197:in `<module:YAML>'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:132:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor/require_vendored.rb:4:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor.rb:40:in `require_libs'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/vendor.rb:53:in `load_vendored'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet.rb:172:in `<module:Puppet>'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet.rb:29:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/lib/puppet/util/command_line.rb:12:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/gems/2.2.0/gems/puppet-3.7.5/bin/puppet:7:in `<top (required)>'
	from /usr/local/bin/puppet:23:in `load'
	from /usr/local/bin/puppet:23:in `<main>'

**Comment by kylog** [comment]
Puppet 4 was released today and is spec-tested against ruby 2.2.0.

**Comment by BanzaiMan** [comment]
Closing this one, since the support is now available.
